control/plugin/metric.go

```diff
@@ -90,8 +90,8 @@ func NewPluginConfigType() ConfigType {
 // Represents a metric type. Only used within plugins and across plugin calls.
 // Converted to core.MetricType before being used within modules.
 type MetricType struct {
-	// Namespace is the identifier for a metric.
-	Namespace_ []core.NamespaceElement `json:"namespace"`
+	// Namespace_ is the identifier for a metric.
+	Namespace_ core.Namespace `json:"namespace"`
```

Comment:
Why did it need to be changed? Is there an example of how it would be used in the plugins? I think we need to be careful/thoughtful about changes that effect plugin authors.